components/odh-notebook-controller/controllers/notebook_webhook.go

```diff
@@ -730,18 +730,28 @@ func SetContainerImageFromRegistry(ctx context.Context, config *rest.Config, not
 					}
 
 					imagestreamFound := false
-					// List imagestreams in the specified namespace
+					// Check for controller namespace before the notebook namespace
+					// to avoid user overriding the image selection in the case of
+					// images having the same name in both namespaces.
+
+					// List imagestreams in the controller namespace
 					imagestreams, err := dynamicClient.Resource(ims).Namespace(namespace).List(ctx, metav1.ListOptions{})
```

Comment:
not to address in this pr, but this listing of imagestreams without pagination will not scale well to thousands of items in single namespace

Reply:
With use of  imagev1 type https://github.com/openshift/api/blob/master/image/v1/types.go
now , look up each imagestream is more convenient and reduces the requirement of pagination.